packages/components/src/navigator/navigator-screen/use-screen-animate-presence.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	useEffect,
+	useRef,
+	useState,
+	useLayoutEffect,
+	useCallback,
+	useMemo,
+} from '@wordpress/element';
+import { usePrevious, useReducedMotion } from '@wordpress/compose';
+import { isRTL as isRTLFn } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import * as styles from '../styles';
+
+const isExitAnimation = ( animationName: string ) =>
+	animationName === styles.slideToLeft.name || styles.slideToRight.name;
+
+export function useScreenAnimatePresence( {
```

Comment:
Lots of good work here, but I'm wondering if we're trying too hard not to use `framer-motion`. When it's about a simple animation, using CSS makes lots of sense. But when we need to reinvent the wheel in cases like animating presence, I feel like the best bet would be to use `framer-motion`. Bundle size is important, but DX and UX are more important, so we might have to make compromises. Especially given that we're looking to build more complex motion and animations for many of the components and flows.

Reply:
I definitely feel _better_ about it after the refactor into a designated hook, but also it’s still unknown how bug-free the current code is, and how many other components will need to animate presence.

My instinct is that it could still make sense to migrate from Framer Motion to a lighter library like React Transition Group. But then again, if it's a substantial effort to migrate from FM to RTG, maybe even that isn't worth it. 

I’d love to hear from @ciampo and @daniguardiola about how they currently assess the overall cost/benefit around these rewrites. Let's make sure the benefits are still big enough to justify our rewrite efforts and potential increase in maintenance cost.

Reply:
As for this particular PR, I'm personally comfortable enough with the logic isolation now to merge it as an incremental improvement over trunk. Is it scalable/sustainable? Maybe not. But it seems like we could swap it out without too much difficulty if we ultimately decide to use a library.